server/filestore.go

```diff
@@ -10906,3 +11109,27 @@ func writeFileWithSync(name string, data []byte, perm fs.FileMode) error {
 	}
 	return f.Close()
 }
+
+// This is to offload UnixNano() processing from timestamp creation for cache management.
+var (
+	tsOnce     sync.Once
+	accessTime atomic.Int64
+)
+
+// Update every 100ms.
+const accessTimeTickInterval = 100 * time.Millisecond
+
+// Will load the access time from an atomic. We will also setup the Go routine
+// to update this in one place.
+func getAccessTime() int64 {
+	tsOnce.Do(func() {
+		accessTime.Store(time.Now().UnixNano())
+		go func() {
+			ticker := time.NewTicker(accessTimeTickInterval)
+			for range ticker.C {
```

Comment:
Doesn't this mean this ticker will keep running in a goroutine for every (file-based) stream, even if the stream itself is idle, not used as much, or maybe even after it's deleted?

Reply:
Yes it will run every 100ms once for any code base pulling it in. I think that is ok, we only fire it up when accessed the first time meaning a filestore stream was spun up.